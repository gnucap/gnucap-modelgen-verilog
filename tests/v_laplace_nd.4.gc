
attach ../vams/vsine.so

verilog

//va_laplace_nd #() a(1,2,3,4);

va_laplace_nd #(.n0(0), .n1(1), .d0(1), .d1(0)) b0(0,1,in,0);
va_laplace_nd #(.n0(0), .n1(1), .d0(1),       ) b0(0,2,in,0);

vsine #(.ampl(1), .freq(1), .phase(-3.14159/2.)) v1(in, 0);
resistor #(1) r1(1,0);
resistor #(1) r1(2,0);

list

print tran v(nodes)
tran 1 .1 basic

print ac vm(nodes) vp(nodes)
ac .01 100 * 10

status notime
