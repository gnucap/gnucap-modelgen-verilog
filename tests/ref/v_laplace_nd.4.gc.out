//va_laplace_nd #() a(1,2,3,4); 
va_laplace_nd #(.n0(0),.n1(1),.d0(1),.d1(0)) b0 (.out0(0),.out1(1),.in0(in),.in1(0));
va_laplace_nd #(.n0(0),.n1(1),.d0(1)) b0 (.out0(0),.out1(2),.in0(in),.in1(0));
vsine #(.freq(1),.ampl(1),.phase(-3.14159/2.)) v1 (.p(in),.n(0));
resistor #(.r(1)) r1 (.p(1),.n(0));
resistor #(.r(1)) r1 (.p(2),.n(0));
#Time       v(1)       v(2)       v(in)     
 0.         0.         0.         1.        
 0.1       -3.7263    -3.7263     0.80902   
 0.2       -6.0278    -6.0278     0.30902   
 0.3       -6.0278    -6.0278    -0.30902   
 0.4       -3.7263    -3.7263    -0.80902   
 0.5       -0.0025001 -0.0025001 -1.        
 0.6        3.7213     3.7213    -0.80902   
 0.7        6.2786     6.2786    -0.30902   
 0.8        6.082      6.082      0.30902   
 0.9        3.7806     3.7806     0.80902   
 1.         0.056736   0.056736   1.        
#Freq       vm(1)      vm(2)      vm(in)     vp(1)      vp(2)      vp(in)    
 0.01       0.062832   0.062832   1.         90.        90.        0.        
 0.1        0.62832    0.62832    1.         90.        90.        0.        
 1.         6.2832     6.2832     1.         90.        90.        0.        
 10.        62.832     62.832     1.         90.        90.        0.        
 100.       628.32     628.32     1.         90.        90.        0.        
Gnucap   System status
iterations: op=0, dc=0, tran=69, fourier=0, total=96
transient timesteps: accepted=21, rejected=1, total=22
nodes: user=3, subckt=0, model=5, total=8
dctran density=62.5%, ac density=62.5%
